src/shared/tpm2-util.c

```diff
@@ -1443,6 +1466,34 @@ void tpm2_tpml_pcr_selection_sub_tpms_pcr_selection(TPML_PCR_SELECTION *l, const
                 tpm2_tpms_pcr_selection_sub(selection, s);
 }
 
+/* Test if all bits in the mask for the hash are set in the TPML_PCR_SELECTION. */
+bool tpm2_tpml_pcr_selection_has_mask(const TPML_PCR_SELECTION *l, TPMI_ALG_HASH hash, uint32_t mask) {
+        TPMS_PCR_SELECTION *s = tpm2_tpml_pcr_selection_get_tpms_pcr_selection((TPML_PCR_SELECTION*) l, hash);
```

Comment:
missing assertion for `l`.

Also, is it not necessary to copy `l`, like `tpm2_tpml_pcr_selection_to_mask()` ?
Or, maybe, it is better to reuse `tpm2_tpml_pcr_selection_to_mask()` ?

Anyway, the cast sounds spurious and dangerous.

Reply:
yep you are correct, i'll update this to use `tpm2_tpml_pcr_selection_to_mask()` instead